### DISCUSSION TITLE
Add Medusa Events to Payment Webhook Events

### DISCUSSION THREAD

**Discussion by fxmb** [medusajs/medusa#5041] [Feature Requests]
Based on the stripe webhook events mentioned [here](https://docs.medusajs.com/plugins/payment/stripe#capture-payments), would it be possible to add medusa events based on those? E.g. I would like to schedule an email to the customer if e.g. a payment failed. So whenever the endpoint for stripe receives this message it would publish an event STRIPE.PAYMENT.FAILED or something and I could subscribe with a basic subscriber in medusa. Currently I would have to overwrite the plugin which I'd rather not.

Is there another way to achieve this?

**Comment by pevey** [comment]
I'm not sure how to add an event, but another way to approach this would be to override the stripe webhooks endpoint with your own.  You could start by duplicating the existing one, and then add your own modifications.  Any route added in src/routes will override routes provided by packages.
